archimedes/archimedes/functions/list/manipulations.py

```diff
@@ -1,16 +1,20 @@
 from collections.abc import Iterable
+import re 
 
 def unique(ls):
     return list(dict.fromkeys(ls))
 
-# def flatten(ls):
-#     return list(itertools.chain.from_iterable(ls))
-
 def flatten(ls):
     def _flatten(l):
         for el in l:
             if isinstance(el, Iterable) and not isinstance(el, (str, bytes)):
                 yield from _flatten(el)
             else:
                 yield el
-    return list(_flatten(ls))
\ No newline at end of file
+    return list(_flatten(ls))
```

Comment:
Yup, this works great.  Unfortunately there isn't a natural built in to do this, I believe.  I'd nit that using recursion here is not great for python in general and is not necessary (you can achieve this with a list and a loop), but honestly I can't imagine the nesting in our data to be deep enough for that to actually matter.

Reply:
Hmmm I think I agree that our data is unlikely to yield deep enough recursion for efficiency here to be the major concern.  My main goal was to have the output be assured to be a completely flat list with no sub-lists, which seems easier with recursion.